engine/CMakeLists.txt

```diff
@@ -1,19 +1,19 @@
-cmake_minimum_required(VERSION 3.7 FATAL_ERROR)
+cmake_minimum_required(VERSION 3.15 FATAL_ERROR)
```

Comment:
Is there any reason why this is the minimal version required? Is it because Ubuntu 20.04 comes with CMake 3.16.3?

Would this be a problem with the Ubuntu 18.04 docker version? If yes shouldn't we officially deprecate this image?

Reply:
Conan CMakeDeps and CMakeToolchain generators require CMake >= 3.15. This has nothing to do with what version an Ubuntu distribution ships. We can inject CMake via Conan profile and this is what we do in this PR to support Ubuntu 18.04.

We *could* let Conan complain about CMake version, but the `cmake` program has some important features gained with 3.15 that we make use of in the Makefiles, so it's best to be explicit here.